Android/Sanku_Yogesh/Beginner/ContactApp/app/src/main/java/android/example/contactapp/customadapter.java

```diff
@@ -0,0 +1,46 @@
+package android.example.contactapp;
+
+import android.content.Context;
+import android.content.Intent;
+import android.os.Parcelable;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.ArrayAdapter;
+import android.widget.ImageView;
+import android.widget.TextView;
+import android.widget.Toast;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+
+public class customadapter extends ArrayAdapter<contact> {
+
+    public contact[] arr;
+    public Context context;
+
+    public customadapter(@NonNull Context context, int resource, @NonNull contact[] arr) {
+        super(context, resource, arr);
+        this.context = context;
+        this.arr = arr;
+    }
+
+
+    @Nullable
+    @Override
+    public contact getItem(int position) {
+        return arr[position];
+    }
+
+    @NonNull
+    @Override
+    public View getView(int position, @Nullable View convertView, @NonNull ViewGroup parent) {
+        convertView = LayoutInflater.from(getContext()).inflate(R.layout.customlayout,parent,false);
```

Comment:
Should use View Holder pattern (use recycled view passed into this method as the second parameter) for smoother scrolling + Dereference of 'getItem(position)' may produce 'NullPointerException' .